src/index.js

```diff
@@ -436,6 +436,7 @@ export var app = function(props) {
   var lock = false
   var state = {}
   var sub = []
+  var middlewareArgs = { action: null, props: null }
```

Comment:
Might make sense for this to be an array, then we could use `middleware.apply(null, middlewareArgs)(state)` a few lines down.